xdsl/printer.py

```diff
@@ -409,6 +409,17 @@ def print_int(self, value: int, type: IntegerType | IndexType | None = None):
         else:
             self.print_string(f"{value:d}")
 
+    def print_dimension_list(self, dims: Sequence[int]):
+        if len(dims) == 0:
+            self.print_string("[]")
+            return
```

Comment:
Let's not print the `[]` here, and use this helper in the other types that print dimensions in this format. The easiest way to find them is to search for "?" in the project.